x-pack/solutions/security/packages/distribution-bar/src/distribution_bar.tsx

```diff
@@ -147,7 +140,20 @@ export const DistributionBar: React.FC<DistributionBarProps> = React.memo(functi
 ) {
   const styles = useStyles();
   const { stats, 'data-test-subj': dataTestSubj, hideLastTooltip } = props;
-  const parts = stats.map((stat) => {
+
+  const [hoveredKey, setHoveredKey] = useState<string | null>(null);
+  const hasCurrentFilter = stats.some((item) => item.isCurrentFilter === true);
```

Comment:
nit:  add `useMemo` to prevent recalculating on every render.:

```suggestion
  const hasCurrentFilter = useMemo(
  () => stats.some((item) => item.isCurrentFilter),
  [stats]
);

```